### PR TITLE
WIP: Support the updated websocket v5 API calls

### LINKED ISSUE #8: OBS WebSocket v5 support
I`ve been using the OBS Message plugin for a long time and now that OBS 28 is released and using WebSocket v5 I`m wondering if the plugin its going to be update also to support this new version

### LINKED ISSUE #9: OBS Message not connecting via OBS Web Socket
OBS version 28.0.3
Elgato Stream Deck version 5.3.3

No additional information available to me.  I've entered the correct IP address, port, and password and it continually fails to connect.  I've verified the web socket settings in OBS and they match what I've entered in Stream Deck.

### PR DESCRIPTION
**Pull request by mrgoodbytes8667**
Needs to support the newer v5 API calls (It does currently connect, but Stream Deck shows it doesn't)

- [x] Verify updated call function actually gets heard by Kruiz Control

### CODE REVIEW

**Comment by mrgoodbytes8667** [comment]
A test build of this pull request for anyone that can assist (will show as version 0.1.4 so that Stream Deck will install it): [live.goodbytes.obsmessage.streamDeckPlugin.zip](https://github.com/mrgoodbytes8667/obsmessage/files/9712564/live.goodbytes.obsmessage.streamDeckPlugin.zip)

**Comment by isopigeon** [comment]
Every time i hit the streamdeck button it seems to reconnect the websocket connection with a close event and no reason. Kruize doesnt seem to react in any way. stream deck shows no connection but OBS does list the connection. 

`11:34:56.316: [obs-websocket] [WebSocketServer::onClose] WebSocket client `[::ffff:127.0.0.1]:50483` has disconnected with code `1000` and reason: 
11:34:56.332: [obs-websocket] [WebSocketServer::onOpen] New WebSocket client has connected from [::ffff:127.0.0.1]:50516`



kruize v1.5.6
obs 28.0.3
streamdeck 5.3.3.15214

**Comment by WiscoKel** [comment]
I also get an error when trying to use this version.

`22:42:40.617: [obs-websocket] [WebSocketServer::onClose] WebSocket client `[::ffff:127.0.0.1]:63697` has disconnected with code `1000` and reason: 
22:42:40.622: [obs-websocket] [WebSocketServer::onOpen] New WebSocket client has connected from [::ffff:127.0.0.1]:62563`

Kruiz Control v1.6.1
OBS 28.0.3
StreamDeck 5.3.3.15214

**Comment by mrgoodbytes8667** [comment]
If I open up the OBS logs I can see where the plugin is sending the message, but KC itself doesn't seem to see it. Using OBS Send itself shows up in the logs but as a warning. I'll have to keep digging.

File: live.goodbytes.obsmessage.sdPlugin/js/obsHandler.js [hunk 1]
```diff
@@ -123,7 +122,7 @@ class ObsHandler {
         if(!this.connected) {
             await this.connect();
         }
-        return this.obs.send('BroadcastCustomMessage', {
+        return this.obs.call('BroadcastCustomEvent', {
```
Comment (Kruiser8) [inline live.goodbytes.obsmessage.sdPlugin/js/obsHandler.js:124]: this needs to be updated to
```js
return this.obs.call('BroadcastCustomEvent', {
  'eventData': {
    'realm': 'kruiz-control',
    'data': {
      'message': message,
      'data': data
    }
  }
})
```

File: live.goodbytes.obsmessage.sdPlugin/js/obsHandler.js [hunk 2]
```diff
@@ -95,14 +95,13 @@ class ObsHandler {
...
+        const options = {};
+        var password = null;
         if (this.password !== '') {
-            options['password'] = this.password;
+            password = this.password;
         }
-        console.log('--- OBS Websocket: connect() ---', options);
-        return this.obs.connect(options);
+        console.log('--- OBS Websocket: connect() ---', this.getOBSAddress(), password, options);
+        return this.obs.connect(this.getOBSAddress(), password, options);
```
Comment (Kruiser8) [inline live.goodbytes.obsmessage.sdPlugin/js/obsHandler.js]: Shouldn't need the `options` input anymore.
```js
return this.obs.connect(this.getOBSAddress(), password);
```

File: live.goodbytes.obsmessage.sdPlugin/js/obsHandler.js [hunk 3]
```diff
@@ -84,7 +84,7 @@ class ObsHandler {
      * @return {string}
```
Comment (Kruiser8) [inline live.goodbytes.obsmessage.sdPlugin/js/obsHandler.js:84]: the default port should now be `4455` (line 54 and 70)

**Comment by mrgoodbytes8667** [comment]
I just applied @Kruiser8's fixes, while I'm still not having any luck specifically here, I'm not 100% sure my Kruiz Control install is setup 100% correct, so I'm going to throw this here for others to test. Again, I'll note that it's going to tell you inside Stream Deck that it doesn't connect, I'm trying to get the actual functionality itself working before I move onto that. Also, I *think* I properly incremented the version here such that it will install on top properly, if not, just comment and I'll fix it.

[live.goodbytes.obsmessage.streamDeckPlugin.zip](https://github.com/mrgoodbytes8667/obsmessage/files/9856915/live.goodbytes.obsmessage.streamDeckPlugin.zip)

**Comment by WiscoKel** [comment]
This new version is working for me

**Comment by mrgoodbytes8667** [comment]
Spun off the remaining two portions so I can "publish" this halfway working build as an alpha.

**Pull request merged**